include/npcomp/Dialect/ATen/ATen.td

```diff
@@ -0,0 +1,161 @@
+//===- ATen.td ---------------------------------------------*- tablegen -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef OP_BASE
+#else
+include "mlir/IR/OpBase.td"
+#endif // OP_BASE
+
+#ifdef ATEN_OPS
+#else
+#define ATEN_OPS
+#endif
+
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "npcomp/Dialect/ATen/ATenOpInterface.td"
+
+def aten_Dialect : Dialect {
+  let name = "aten";
+  let cppNamespace = "aten";	
+}
+
+class aten_Op<string mnemonic, list<OpTrait> traits = [StatisticsOpInterface]> :
+    Op<aten_Dialect, mnemonic, traits>;
+
+
+// Most ops are automatically generated from pytorch specs.
+include "npcomp/Dialect/ATen/ATenOps.td"
+
+
+def aten_BatchNormOp: aten_Op<"batch_norm", [NoSideEffect, StatisticsOpInterface]>,
+                      Results<(outs AnyTensor:$output, AnyTensor:$save_mean, AnyTensor:$save_invstd)> {
+  let arguments = (
+    ins AnyType:$arg0,
+        AnyType:$arg1,
+        AnyType:$arg2,
+        AnyType:$arg3,
+        AnyType:$arg4,
+        AnyType:$arg5,
+        AnyType:$arg6,
+        AnyType:$arg7,
+        AnyType:$arg8
+  );
+
+  let summary = "BatchNorm operator";
+  let description = [{
+    BatchNorm operator
+  }];
+  let extraClassDeclaration = [{
+    std::map<std::string, uint64_t> getStatistics();
+  }];
+}
+
+// We have list constants, which come out of pytorch.  Represent them using
+// our own constant-like type, which gets lowered to std_ConstantOp later.
+def aten_ConstantOp: aten_Op<"constant", [NoSideEffect]>,
+                     Results<(outs AnyType)> {
+  let summary = "Constant operator";
+  let description = [{
+    Constant operator
+  }];
+
+}
+
+// Our jit library only supports 6 argument convolutions, rather than 9
+// arguments supported by pytorch.  This operation allows us to represent this
+// limitation temporarily.
+def aten_ConvolutionOp: aten_Op<"_convolution", [NoSideEffect, StatisticsOpInterface]>,
+                        Results<(outs AnyTensor)> {
+  let arguments = (
+    ins AnyTensor:$input,
+        AnyTensor:$weight,
+        AnyTensor:$bias,
+        AnyType:$stride,
+        AnyType:$padding,
+        AnyType:$dilation
+  );
+
+  let summary = "Convolution operator";
+  let description = [{
+    Convolution operator
+  }];
```

Comment:
In general the doc for these ops is a bit terse right now, can you revisit these? Possibly linking to an official documentation page would be fine as well I guess.

Reply:
moved to issue.  Most of these are automatically generated, so the generator needs to get updated.